src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -1934,9 +1934,36 @@ private class OffsetAndRule {
 
 
 #if PLATFORM_UNIX
-        static public TimeZoneInfo FindSystemTimeZoneById(string id) {
-            // UNIXTODO
-            throw new NotImplementedException();
+        static public TimeZoneInfo FindSystemTimeZoneById(string id)
+        {
+            if (string.IsNullOrEmpty(id))
+            {
+                throw new ArgumentNullException("id");
+            }
+
+            // Special case for Utc since there is code that expects only 1 instance of the UTC time zone info to be created.
+            // see CachedData.GetCorrespondingKind()
+            if (String.Compare(id, c_utcId, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                return TimeZoneInfo.Utc;
+            }
+
+            // UNIXTODO: use CachedData
```

Comment:
> Is there an issue tracking this?

I was going to do this in a future change for this same general "Globalization on Linux" Issue 846.
I can log a separate Issue, if that's the standard procedure.  It's totally up to you.